File: 976.largest-perimeter-triangle/solution.py
class Solution:
    def largestPerimeter(self, nums: List[int]) -> int:
        nums.sort(reverse = True)

        N = len(nums)
        A, B = nums[0], nums[1]
        for i in range(2, N):
            C = nums[i]

            if (A + B > C) and (B + C > A) and (A + C > B):
                return A + B + C
            A, B = B, C
        return 0

File: 976.largest-perimeter-triangle/solution.kt
class Solution {
    fun largestPerimeter(nums: IntArray): Int {
        nums.sortDescending()
        val N = nums.size

        var A = nums[0]
        var B = nums[1]
        var C = 0
        for (i in 2..N - 1) {
            C = nums[i]
            if ((A + B > C) && (B + C > A) && (A + C > B)) {
                return A + B + C
            }
            A = B
            B = C
        }

        return 0
    }
}
